geoprocessing/src/modules/worker/worker-resolver.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable, OnModuleDestroy, Scope } from '@nestjs/common';
+import { Job, Worker } from 'bullmq';
+import { Config } from './config';
+import { WorkerProcessor } from './worker-processor';
+
+@Injectable({
+  scope: Scope.TRANSIENT,
+})
+export class WorkerResolver implements OnModuleDestroy {
+  private _worker?: Worker;
+
+  constructor(private readonly config: Config) {}
+
+  wrap<Input, Output>(
+    queueName: string,
+    processor: WorkerProcessor<Input, Output>,
+  ): Worker<Input, Output> {
+    if (this._worker) {
+      throw new Error('Worker is already created!');
+    }
+    this._worker = new Worker<Input, Output>(
+      queueName,
+      (job: Job) => processor.process(job),
+      this.config.redis,
+    );
+    return this._worker as Worker<Input, Output>;
```

Comment:
```suggestion
    const worker = new Worker<Input, Output>(
      queueName,
      (job: Job) => processor.process(job),
      this.config.redis,
    );
    this._worker = worker;
    return worker;
```
no cast needed :)

Reply:
fixed in https://github.com/Vizzuality/marxan-cloud/pull/202/commits/d4a3c87a04836695481fb7e64cc6eaa76d5bdcb2